src/Connection.php

```diff
@@ -236,48 +255,56 @@ protected static function getPDO(array $dsn)
      */
     public function dsql($properties = []): Query
     {
-        $c = $this->query_class;
-        $q = new $c($properties);
-        $q->connection = $this;
+        $query = new $this->queryClass($properties);
+
+        $query->connection = $this->handler();
 
-        return $q;
+        return $query;
     }
 
     /**
      * Returns Expression object with connection already set.
      *
      * @param string|array $properties
-     * @param array        $arguments
+     * @param array        $args
      */
-    public function expr($properties = [], $arguments = null): Expression
+    public function expr($properties = [], $args = null): Expression
     {
-        $c = $this->expression_class;
-        $e = new $c($properties, $arguments);
-        $e->connection = $this->connection ?: $this;
+        $expression = new $this->expressionClass($properties, $args);
+
+        $expression->connection = $this->handler();
 
-        return $e;
+        return $expression;
     }
 
     /**
-     * Returns Connection or PDO object.
+     * @deprecated use Connection::handler instead
+     */
+    public function connection()
+    {
+        return $this->handler();
+    }
+
+    /**
+     * Returns the connection handler.
      *
      * @return Connection|\PDO
      */
-    public function connection()
+    public function handler()
```

Comment:
Driver seems more industry common to me here.

Is it desireable to allow - use case? - `\Connection` here at all?

Reply:
Using `handler` here also needed some thought. Handler can be a `Connection` and `PDO` or some other class so it has use cases and does not confuse having `Connection::connection`
Handler handles the connection

Reply:
I got confused myself in the past... Ok to rename to `driver`?

Reply:
Renamed to driver now if it is more understandable